#1324 Permissions patch to separate permssions

GregSutcliffe opened the pull request:
Permissions patch to separate permssions on hosts from permissions for objects within hosts. This allows a user to be granted permission to edit the host (and so change the group or proxy) but not, for example, edit the parameters

This could probably be extended further if necessary.

Refs #1324

Commit:
Permissions patch to separate permssions on
 hosts from permissions for objects within
 hosts. This allows a user to be granted
 permission to edit the host (and so change the
 group or proxy) but not, for example, edit the
 parameters

This could probably be extended further if necessary.

Refs #1324

ohadlevy commented:
I've started to amend the patch, so far here is what i did: http://pastie.org/3070707.

but in anycase, it seems that a few unit tests are failing, so I kind of stopped at that point.

thanks!

ohadlevy commented:
bump!

GregSutcliffe commented:
That pastie looks sane, and far cleaner than my first cut. I really should thing of those kind of tricks :)

Which tests are failing? As I recall (ok, it was last year :P) the only failing tests I had were ones which failed before I made any changes.

Commit:
Ensure a User with hostgroup restrictions cannot View or Edit
other Hostgroups.

Ties in with my earlier patch to alter permissions - a User must
have edit_classes to change the classes in a hostgroup, and
must have create/edit/destroy_parameters to alter that part of
the Hostgroup page. Edit Hostgroups suffices for changing things like
the name and the proxy.

Fixes #1484 and Refs #1324

GregSutcliffe commented:
Ready to go, full test pass :)

ohadlevy commented:
seems good of first review, mind merging the code from my pastie as well?

thanks!

GregSutcliffe commented:
I've merged that in, but I'm getting some odd test failures. I'm pretty sure it's the tests that are wrong (rather than the code) but I cannot seem to work out why. 

I'd love to grab say an hour of your time sometime soon to do a G+ hangout where I can share my screen with you, show you the test failures, and get some help on how to fix it - otherwise I don't think I'd going to be able to. I've spent 4 hours on it today already :P

EDIT: I've pushed the pastie stuff to this branch if you want to try it yourself.

Commit:
Add Ohad's review comments and fix tests

One fixture was broken by specifying the :id - fixed
One legacy test for the :query interface (removed in 17637a7)
has been removed.
